### ISSUE TITLE
Can't set middlewares on RouteGroup if an Resource is inside an Group

### ISSUE THREAD

**Issue by rfschubert** [MasoniteFramework/api#8]
I refatored my routes to use `RouteGroup`.

This is my current refactored `web.py` that fires the bug.

```python
"""Web Routes."""
from masonite.routes import Get, Post, RouteGroup

from app.resources.DocumentResource import DocumentResource
from app.resources.RejectionReasonResource import RejectionReasonResource

ROUTES = [
    RouteGroup([
        RouteGroup([
            RejectionReasonResource('/rejection-reason').routes(),
            Get().route('/get_from_array', 'DocumentController@show').name('get_from_array'),
            Get().route('/pending-validation', 'DocumentController@list_pending_validation').name('list_pending_validation'),
        ], prefix='/document'),
        DocumentResource('/document').routes(),
    ], prefix='/api')
]
```

It explodes this error

```shell
Serving at: http://127.0.0.1:8010
/Users/raphaelschubert/projects/digitalbanks/sibyl/routes/web.py changed; reloading ...
Killing server with PID 19119.
Starting monitor for PID 19131.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/hupper/ipc.py", line 315, in spawn_main
    func(**kwargs)
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/hupper/worker.py", line 267, in worker_main
    func(*spec_args, **spec_kwargs)
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/commands/_devserver.py", line 129, in run
    wsgi_handler = _import_application(module_name, app_name)
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/commands/_devserver.py", line 122, in _import_application
    module = import_module(module_name)
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/raphaelschubert/projects/digitalbanks/sibyl/wsgi.py", line 6, in <module>
    from config import application, providers
  File "/Users/raphaelschubert/projects/digitalbanks/sibyl/config/providers.py", line 4, in <module>
    from masonite.providers import (AppProvider, BroadcastProvider, CacheProvider,
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/providers/__init__.py", line 1, in <module>
    from .AppProvider import AppProvider
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/providers/AppProvider.py", line 23, in <module>
    from routes import web
  File "/Users/raphaelschubert/projects/digitalbanks/sibyl/routes/web.py", line 16, in <module>
    ], prefix='/api', middleware='csrf')
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/routes.py", line 458, in __new__
    self._middleware(self, *middleware)
  File "/Users/raphaelschubert/.local/share/virtualenvs/sibyl-7vpPK4zW/lib/python3.7/site-packages/masonite/routes.py", line 481, in _middleware
    route.middleware(*middleware)
AttributeError: 'RejectionReasonResource' object has no attribute 'middleware'
Server with PID 19131 exited with code 1.
```

So i can't configure middlewares to all routes inside an `RouteGroup` if it has any `Resource` nested on it.

**Comment by rfschubert** [comment]
Routes with @params will have `csrf` issues, issue opened at
https://github.com/MasoniteFramework/core/issues/574

**Comment by josephmancuso** [comment]
these should both be fixed in version 0.0.4
